Contents/Artifacts/FateDeciderArtifact.cs

```diff
@@ -13,204 +13,163 @@
-                    {
-                        Main.projectile[proj].aiStyle = -1;
-                        Main.projectile[proj].ai[0] = -1;
-                        Main.projectile[proj].ai[1] = AmmoID.Arrow;
-                    }
-                }
-            }
-            if (item.useAmmo == AmmoID.Bullet)
-            {
-                if (GoodBuffIndex == 1)
-                {
-                    int proj = Projectile.NewProjectile(source, position, velocity.Vector2RotateByRandom(10) * .25f, type, damage, knockback, Player.whoAmI);
-                    if (Main.projectile[proj].ModProjectile is null)
-                    {
-                        Main.projectile[proj].aiStyle = -1;
-                        Main.projectile[proj].ai[0] = -1;
-                        Main.projectile[proj].ai[1] = AmmoID.Bullet;
-                    }
-                }
-            }
-            if (!item.noMelee && !item.noUseGraphic && item.DamageType == DamageClass.Melee)
-            {
-                if (GoodBuffIndex == 2)
-                {
-                    for (int i = 0; i < 3; i++)
-                    {
-                        Vector2 newpos = position.PositionOFFSET(velocity.Vector2DistributeEvenly(3, 45, i), -90);
-                        int proj = Projectile.NewProjectile(source, newpos, (Main.MouseWorld - newpos).SafeNormalize(Vector2.Zero) * item.shootSpeed, type, damage, knockback, Player.whoAmI);
-                        if (Main.projectile[proj].ModProjectile is null)
-                        {
-                            Main.projectile[proj].penetrate = 1;
-                        }
-                    }
-                }
-            }
-            if (item.mana > 0 && item.DamageType == DamageClass.Magic)
-            {
-                if (GoodBuffIndex == 3)
-                {
-                    Vector2 newpos = Main.rand.NextVector2Circular(100, 100);
-                    Projectile.NewProjectile(source, newpos, (Main.MouseWorld - newpos).SafeNormalize(Vector2.Zero) * item.shootSpeed, type, damage, knockback, Player.whoAmI);
-                }
-                if (BadBuffIndex == 3)
-                {
-                    Player.statLife -= item.mana;
-                }
-            }
-            return base.Shoot(item, source, position, velocity, type, damage, knockback);
-        }
-        public override void OnHitNPCWithItem(Item item, NPC target, NPC.HitInfo hit, int damageDone)
-        {
-            base.OnHitNPCWithItem(item, target, hit, damageDone);
-            if (GoodBuffIndex == 5)
-            {
-                Vector2 pos = target.Center - Main.rand.NextVector2CircularEdge(100, 100);
-                Projectile.NewProjectile(Player.GetSource_ItemUse(item), pos, (target.Center - pos).SafeNormalize(Vector2.Zero) * 3f, ModContent.ProjectileType<ArtifactSwordSlashProjectile>(), item.damage, 0, Player.whoAmI);
-            }
-        }
-    }
+	class FateDeciderPlayer : ModPlayer {
```

Comment:
Please prefer being explicit about accessibility.
Usually you would want `public` in most cases.

Reply:
alright